### PR TITLE
adiv5 component probe cleanup

### PR DESCRIPTION
**Pull request by perigoso**
This is mostly formatting cleanups and code styling for readability, but there are a couple more significant changes in the mix, in particular the "aliasing" of a couple of "designer ids" that consolidate the STM probe methods, that warrants some testing

The main motivation was me trying and failing to understand the code for a bit and having to delve deeper, there were some magic offsets (it's documented, just somewhere else) and masks and duplicated code when dealing with the designer code

I tested this on a couple of Renesas devices (was working on support, soon TM), and everything related with this code behaved as expected.

### CODE REVIEW

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:421]: Please clarify what this case was and how it's now handled

File: src/target/cortexm.c
Comment (dragonmux) [inline src/target/cortexm.c:415]: Please clarify what this case was and how it's now handled

File: src/target/adiv5.h
Comment (dragonmux) [inline src/target/adiv5.h:146]: This is good and a much needed bit of cleanup, however it would be useful in-context if "DESIGNER" or "MANUFACTURER" was somewhere in the constant name - eg `JEP106_DESIGNER_ARM` to provide a stronger association to these definitely being `ap_designer` codes. This also suggests that `ap_designer` should be renamed to `ap_jep106_designer` or similar

File: src/target/adiv5.h
Comment (dragonmux) [inline src/target/adiv5.h:92]: clang-tidy emits warnings when the U is lower case - please convert to upper case as you're doing the big reformat on these files.

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:519]: Can this be made const?

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:484]: It might help to add a comment explaining why this is necessary so people aren't tempted to "fix" it or fiddle with it without understanding the full consequences of doing so.

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:422]: please rename this timeout to something a little more informative

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:370]: Please add 'U' on the end of the constant for safety.

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:328]: Please give this timeout a better name

File: src/target/adiv5.c
Comment (dragonmux) [inline src/target/adiv5.c:136]: It's not for this PR, but at some point we should make this depend on ADIv5_DEBUG rather than the general debug define, so we can enable and disable detailed ADIv5 decoder logic debug separately in the menuconfig in v2.

**Review by dragonmux — CHANGES_REQUESTED** [review]
This is looking really good - there are just a few items we spotted that could do with clarifications or explanation before we're happy to merge it, but it's already a massive improvement over what was there. Thank you :)

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c:415]: the context of this removal is [here](https://github.com/blackmagic-debug/blackmagic/blob/f16fe54d24c3278378a8651f7e777dd5ad0f0c13/src/target/adiv5.h#L159-L163)

now this "AP Designer" is aliased to STM, and the case that handles these devices follows

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c:421]: the context of this removal is [here](https://github.com/blackmagic-debug/blackmagic/blob/f16fe54d24c3278378a8651f7e777dd5ad0f0c13/src/target/adiv5.h#L165-L178)

now this "AP Designer" is aliased to STM, and the case that handles these devices follows

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c:415]: sorry, this was supposed to direct you to the correct lines in the commit, but github did not collaborate, follow the lines in the link

File: src/target/cortexm.c
Comment (perigoso) [reply src/target/cortexm.c:421]: sorry, this was supposed to direct you to the correct lines in the commit, but github did not collaborate, follow the lines in the link

Comment (dragonmux) [reply src/target/cortexm.c:421]: That works. Thank you :)

File: src/target/adiv5.c
Comment (perigoso) [reply src/target/adiv5.c:136]: noted, that sounds like a good idea

File: src/target/adiv5.h
Comment (perigoso) [reply src/target/adiv5.h:146]: hm, the adiv5 spec does name it designer though, ugh

Comment (perigoso) [reply src/target/adiv5.h]: I'm going to keep `designer` internally for variables and so, but name the macros `JEP106_MANUFACTURER_xxx`, they don't tie as neatly this was, but they both better align with their respective standards

**Comment by perigoso** [comment]
should be ready, I'd appreciate some testing, as I did not have anything on hand to test this with the last changes, and with the amount of back and forth I did I'm afraid I messed something up

**Review by dragonmux — COMMENTED** [review]
This is looking great! We'll dive into how to test it over the next couple of days with the intent to merge shortly after.

File: src/target/lpc11xx.c
Comment (perigoso) [reply src/target/lpc11xx.c:175]: From what i gathered, it's only used here and i a couple bmp commands

**Review by dragonmux — APPROVED** [review]
LGTM, we'll merge once we've tested it with a few targets to make sure as best we can that nothing broke too badly

**Review by dragonmux — APPROVED** [review]
Tested with a couple of different targets which all still worked properly. Merging.

**Pull request merged**
